static/extensions/LordCat0/ProjectInterfaces.js

```diff
@@ -0,0 +1,442 @@
+        element.style.pointerEvents = 'auto'
+        element.style.userSelect = 'none'
+        element.style.color = 'black'
+        if(args.type == 'Image'){element.draggable = false}
+        elements[args.id] = element
+        elementbox.append(element)
+        metadata[args.id] = {x: 0, y: 0, direction: 90, width: boundingRect.width, height: boundingRect.height, hovered: false, clicked: false}
+        this.FixPos(args.id)
+        element.addEventListener("mouseover", () => metadata[args.id].hovered = true)
+        element.addEventListener("mouseout", () => metadata[args.id].hovered = false)
+        element.addEventListener("click", () => metadata[args.id].clicked = true)
+    }
+    Position(args){
+        if(!elements[args.id]){return}
+        const element = elements[args.id]
+        element.style.left = `${(vm.runtime.stageWidth/2) + args.x - (element.offsetWidth/2)}px`
+        element.style.top = `${(vm.runtime.stageHeight/2) - args.y - (element.offsetHeight/2)}px`
+        metadata[args.id].x = args.x
+        metadata[args.id].y = args.y
+    }
+    Direction(args){
+        if(!elements[args.id]){return}
+        const element = elements[args.id]
+        element.style.transform = `rotate(${args.dir - 90}deg)`
+        metadata[args.id].direction = args.dir
+    }
+    Scale(args){
+        if(!elements[args.id]){return}
+        const element = elements[args.id]
+        element.style.width = `${args.width}px`
+        element.style.height = `${args.height}px`
+        metadata[args.id].width = args.width + 'px'
+        metadata[args.id].height = args.height + 'px'
+        element.style.objectFit = 'fill'
+        this.FixPos(args.id)
+    }
+    Layer(args){
+        if(!elements[args.id]){return}
+        const element = elements[args.id]
+        element.style.zIndex = args.layer
+    }
+    Cursor(args){
+        if(!elements[args.id]){return}
+        const element = elements[args.id]
+        console.log(args.cursor)
+        element.style.cursor = args.cursor
+    }
+    Color(args){
+        if(!elements[args.id]){return}
+        if(elements[args.id].tagName == 'DIV'){elements[args.id].style.backgroundColor = args.color; return} // for people who are confused why 'color' doesnt work with div/box elements
+        elements[args.id].style.color = args.color
+    }
+    BackgroundColor(args){
+        if(!elements[args.id]){return}
+        elements[args.id].style.backgroundColor = args.color
+    }
+    CustomCSS(args){
+        if(!elements[args.id]){return}
+        const element = elements[args.id]
+        let style = document.getElementById(`LCGuiStyle_${args.id}`)
+        let lines = args.css.split(";")
+        if(!style){
+            style = document.createElement('style')
+            style.id = `LCGuiStyle_${args.id}`
+            document.head.append(style)
+        }
+        style.textContent = `[data-id='${args.id}']{\n${lines.join(" !important;\n") + " !important"}\n}`
```

Comment:
possible but unlikely security risk with older browsers allowing embedded javascript inside css

Reply:
oh this isnt supposed to be part of the review